superset/utils/pandas_postprocessing.py

```diff
@@ -264,6 +264,16 @@ def pivot(  # pylint: disable=too-many-arguments
     #  Remove once/if support is added.
     aggfunc = {na.column: na.aggfunc for na in aggregate_funcs.values()}
 
+    # When dropna = False, the pivot_table function will calculate cartesian-product
+    # for MultiIndex.
+    # https://github.com/apache/superset/issues/15956
+    # https://github.com/pandas-dev/pandas/issues/18030
+    series_set = set()
+    if not drop_missing_columns and columns:
+        for row in df[columns].itertuples():
+            metrics_and_series = list(aggfunc.keys()) + list(row[1:])
+            series_set.add("".join([str(_) for _ in metrics_and_series]))
```

Comment:
Let's put a character in the join to remove any risk of collisions. Something like `"_".join([str(_) for _ in metrics_and_series])` should be enough.

Reply:
done